src/qt/frame.cpp

```diff
@@ -115,20 +115,25 @@ void wxFrame::SetStatusBar( wxStatusBar *statusBar )
 
 void wxFrame::SetToolBar(wxToolBar *toolbar)
 {
-    int area = 0;
     if ( toolbar != NULL )
     {
-        if (toolbar->HasFlag(wxTB_LEFT))   area |= Qt::LeftToolBarArea;
-        if (toolbar->HasFlag(wxTB_RIGHT))  area |= Qt::RightToolBarArea;
-        if (toolbar->HasFlag(wxTB_TOP))    area |= Qt::TopToolBarArea;
-        if (toolbar->HasFlag(wxTB_BOTTOM)) area |= Qt::BottomToolBarArea;
+        if ( m_frameToolBar != NULL )
+        {
+            GetQMainWindow()->removeToolBar(m_frameToolBar->GetQToolBar());
```

Comment:
Sorry, this doesn't seem right. We shouldn't call `removeToolBar()` if the new toolbar is non-null.